python/lsst/ts/externalscripts/auxtel/latiss_acquire.py

```diff
@@ -0,0 +1,604 @@
+        the imageInOODS event.
+
+        Parameters
+        ----------
+        timeout: `float`
+            Amount of time to wait for image to arrive.
+        """
+
+        self.log.debug(
+            f"Waiting for image to arrive in OODS for a maximum of {self.image_in_oods_timeout} seconds."
+        )
+        in_oods = await self.latiss.rem.atoods.evt_imageInOODS.next(
+            timeout=self.image_in_oods_timeout, flush=flush
+        )
+
+        data_id = parse_obs_id(in_oods.obsid)
+        self.log.info(f"seq_num {data_id['seq_num']} arrived in OODS")
+
+        return data_id
+
+    async def find_offset(self, data_id, target_position):
+        """Detects the brightest star in an image and returns the offsets
+        (dx_arcsec and dy_arcsec) required to move the target to the target
+        position.
+
+        Parameters
+        ----------
+        data_id: `dict`
+            data Id for initial exposure with target initial position.
+        target_position: `array`
+            desired final position in detector coordinates as array (X,Y).
+
+        Returns
+        -------
+        dx_arcsec, dy_arcsec: `float`, `float`
+            offsets (arcsec) required to translate target from detected
+            position to desired final position
+        """
+        exp = await get_image(
+            data_id,
+            self.best_effort_isr,
+            timeout=self.acq_exposure_time + STD_TIMEOUT,
+        )
+
+        # Find brightest star in image.
+        loop = asyncio.get_event_loop()
+        executor = concurrent.futures.ThreadPoolExecutor()
+        result = await loop.run_in_executor(executor, self.qm.run, exp)
+
+        # Verify a result was achieved, if not raise the exception.
+        if not result.success:
+            raise RuntimeError("Centroid finding algorithm was unsuccessful.")
+
+        current_position = PointD(
+            result.brightestObjCentroid[0], result.brightestObjCentroid[1]
+        )
+
+        # Find offsets to desired position
+        self.log.debug(
+            f"Current brightest target position is {current_position} whereas the "
+            f"target position is {self.target_position}"
+        )
+
+        dx_arcsec, dy_arcsec = calculate_xy_offsets(current_position, target_position)
+        return dx_arcsec, dy_arcsec
+
+    async def execute_acquisition(self, target_position):
+        """Perform acquisition of target. Will iteratively measure position of
+        brightest object in image, calculate offsets required to translate to
+        target_position, apply offsets, take a new image, measure position of
+        brightest object in new image and check that it has arrived at target
+        position within tolerance. Iterations will continue until either the
+        brightest object has arrived at position within tolerance or
+        configurable max acq iters parameter is met.
+
+        Parameters
+        ----------
+        target_position: `array`
```

Comment:
Instead of receiving an array, it would probably be best to take x/y values.

Reply:
Okay, I made the parameters explicit here and elevated it to the configuration stage. This propagated through the script and unit tests, so I put this change is on it's own commit.